nixos/modules/services/desktops/tumbler.nix

```diff
@@ -0,0 +1,50 @@
+# Tumbler
+
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.tumbler;
+  tumbler = cfg.package;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.tumbler = {
+
+      enable = mkEnableOption "Tumbler, A D-Bus thumbnailer service";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.xfce4-13.tumbler;
+        description = "Which tumbler package to use";
+        example = pkgs.xfce4-12.tumbler;
+      };
+
+    };
+
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [
+      tumbler
```

Comment:
There are plug-ins in `/lib/tumbler-1/plugins` but those would not really be looked for and some kind of config file in `/etc/xdg/tumbler`, not sure if that will even get linked.

Reply:
I think tumbler uses `g_get_system_config_dirs` to find the config file. Looked in a vm I see `/run/current-system/sw/etc/xdg/tumbler` linked.

Reply:
Yeah, then that would be [added](https://github.com/NixOS/nixpkgs/blob/2bbb623c19758ecf03d2db20339e4bfb1df90531/nixos/modules/programs/environment.nix#L41) to the environment variable, so the config file is probably the reason for being in `systemPackages`.